fixed icon case sensitivity + new feature + further changes

JoelSjogren opened the pull request:
(i'm using linux)

JoelSjogren committed:
fixed icon case sensitivity

JoelSjogren committed:
added pruning functionality

Pruning means that the page is edited to remove unwanted languages entirely!

For technical reasons, this only works in the English version of Wiktionary.

JoelSjogren commented:
new feature

DavidPx commented:
Thanks for the contribution, Joel, this is good stuff!

I'll take a look at this in the next few days.

JoelSjogren committed:
disable pruning for Appendix pages

e.g. https://en.wiktionary.org/wiki/Appendix:Variations_of_%22du%22

otherwise it shows up as completely empty!

Commit:
changed to scroll to relevant header

Commit:
Incorporated Joel's changes

Commit:
updated readme

DavidPx commented on files/background.js:
```diff
@@ -44,6 +44,20 @@ chrome.runtime.onInstalled.addListener(function() {
+	getSavedItem(wiktionaryFilterPrune, (prune) => {
+	    if (!prune) return;
+	    getSavedItem(wiktionaryFilterPruneExcept, (pruneExcept) => {
+		let pruneExceptList = pruneExcept.split(', ');
```
The split needs to be better about spacing; try this:

```suggestion
let pruneExceptList 
              = pruneExcept
                .split(',')
                .map((s) => s.trim());
```

I was able to break the pruning by putting in `Foo,Bar`.

DavidPx commented:
Cool feature, @JoelSjogren!  I just got it integrated into my code.

One problem:

* If the pruned languages are not found then it removes all content - I feel that it should do nothing in that scenario. 


The pruning feature is great!  Thank you for spending time on that - this is what I envisioned for the extension but was too lazy to implement.  I do see jumping & pruning as exclusive modes, and have implemented this in the [v2 branch](https://github.com/DavidPx/WiktionaryLanguageFilter/tree/v2).  Could you please pull that down and give it a try?

JoelSjogren commented:
> If the pruned languages are not found then it removes all content - I feel that it should do nothing in that scenario.

I agree. I added that exceptional "Appendix:" case and this would be another.

> I do see jumping & pruning as exclusive modes, and have implemented this in the [v2 branch](https://github.com/DavidPx/WiktionaryLanguageFilter/tree/v2). Could you please pull that down and give it a try?

I don't like this. I prefer having both enabled at the same time.

JoelSjogren committed:
avoid clearing the whole page

fixes https://github.com/DavidPx/WiktionaryLanguageFilter/pull/1#issuecomment-778675777

JoelSjogren committed:
improve the handling of exceptional cases

When pruning is avoided (because no results match), print a
message explaining this at the top of the page. Otherwise this
'non-monotone' behavior might be confusing to users.

Also, say you are reading the etymology of some word, and you click
on a link to a word in a language that would normally be pruned,
adding this language to the URL like '#Greek'. Assume furthermore
that this word does exist in some other non-pruned language. Then,
at this page, 'Greek' should not be pruned. Otherwise the link
would be malfunctioning.

DavidPx commented:
Cool, I can see the purpose of having both features running at once.

I'm merging in your work and will remove the dropdown list.  Thank you for your contribution!
